checkov/secrets/plugins/custom_regex_detector.py

```diff
@@ -43,26 +50,84 @@ def analyze_line(
+                is_multiline=True,
+                **kwargs
+            )
+
+        return output
+
+    def _find_potential_secret(
+            self,
+            filename: str,
+            string_to_analyze: str,
+            output: Set[PotentialSecret],
+            line_number: int = 0,
+            context: Optional[CodeSnippet] = None,
+            is_multiline: bool = False,
+            **kwargs: Any
+    ) -> None:
+        current_denylist: Set[Pattern[str]] = self.multiline_denylist if is_multiline else self.denylist
+        current_regex_to_metadata: dict[str, dict[str, Any]] = self.multiline_regex_to_metadata if is_multiline else self.regex_to_metadata
+        for match, regex in self.analyze_string(string_to_analyze, current_denylist, **kwargs):
             try:
                 verified_result = call_function_with_arguments(self.verify, secret=match, context=context)
                 is_verified = True if verified_result == VerifiedResult.VERIFIED_TRUE else False
             except Exception:
                 is_verified = False
-            regex_data = self.regex_to_metadata[regex.pattern]
-            ps = PotentialSecret(type=regex_data["Name"], filename=filename, secret=match,
-                                 line_number=line_number, is_verified=is_verified)
-            ps.check_id = self.regex_to_metadata[regex.pattern]["Check_ID"]  # type:ignore[attr-defined]
-            if len(cast(str, ps.secret_value)) in range(MIN_CHARACTERS, MAX_CHARACTERS) or not regex_data['isCustom']:
+            regex_data = current_regex_to_metadata[regex.pattern]
+            # Wrap multiline match with fstring + ''
+            match = f"'{match}'" if is_multiline else match
+            ps = PotentialSecret(
+                type=regex_data["Name"],
+                filename=filename,
+                secret=match,
+                line_number=line_number,
+                is_verified=is_verified
+            )
+            ps.check_id = current_regex_to_metadata[regex.pattern]["Check_ID"]  # type:ignore[attr-defined]
+            if is_multiline:
+                output.add(ps)
+            elif len(cast(str, ps.secret_value)) in range(MIN_CHARACTERS, MAX_CHARACTERS) or not regex_data['isCustom']:
                 output.add(ps)
             else:
                 logging.info(
                     f'Finding for check {ps.check_id} are not 5-100 characters in length, was ignored')  # type: ignore
 
-        return output
-
-    def analyze_string(self, string: str, **kwargs: Optional[Dict[str, Any]]) -> Generator[Tuple[str, Pattern[str]], None, None]:  # type:ignore[override]
-        for regex in self.denylist:
+    def analyze_string(self, string: str, regex_denylist: Optional[Set] = None, **kwargs: Optional[Dict[str, Any]]) -> Generator[Tuple[str, Pattern[str]], None, None]:  # type: ignore # type:ignore[override]
+        if regex_denylist is None:
+            regex_denylist = self.denylist
```

Comment:
@pazbechor Please fetch the `regex_denylist` from the kwargs, and then create a local list unifying the self.denylist with the regex_denylist. I don't want to break signatures

Reply:
pls check if what I did is fine :)